Add Avatar and ImageInput components (#922)
sumup-oss/circuit-ui: packages/circuit-ui/components/Avatar/Avatar.docs.mdx

import { Status, Props, Story } from '../../../../.storybook/components';

# Avatar

<Status.Stable />

The Avatar component is used to display an avatar or an object image. It is used internally by the AvatarInput component, to allow users to upload images.

<Story id="components-avatar--base" />

<Props />

## Usage guidelines

- **Do** use the right variant for your use case (see _Variants_ below).
- **Do** use the AvatarInput component to allow users to upload images.

> - **Do not** add alt text if the Avatar is purely presentational (for example to show a thumbnail next to a user or product name).

Author: I added a separated Accessibility section, you can see it in the deployed Storybook preview for review 🙂

import { Status, Props, Story } from '../../../../.storybook/components';

# Avatar

<Status.Stable />

The Avatar component displays an identity or an object image. It can be passed to the [ImageInput](Forms/ImageInput) to allow users to upload an avatar.

<Story id="components-avatar--base" />

<Props />

## Usage guidelines

- **Do** use the right variant for your use case (see _Variants_ below).

> - **Do** use the [ImageInput component](Forms/ImageInput) with the `component={Avatar}` prop to allow users to upload an avatar (note: the ImageInput only supports `variant="object"` for now).

Reviewer: What's the reason for the `variant="object"` limitation?

Author: (same as [above](https://github.com/sumup-oss/circuit-ui/pull/922#discussion_r643464874))

Reviewer: Hm, that's annoying. I'll try to think of a solution 🤔

Author: It's also not mandatory for v3, it can be improved as a minor 🙂

Reviewer: Let's keep it like this for now and think of a solution when we need it. As you said, this can be changed in a minor.

> ## Accessibility

Reviewer: Thanks for adding a dedicated _Accessibility_ section! 🙌🏻

Some feedback: I would flip the order of the recommendations: adding alt text should be the default, omitting it the exception. It's better to have duplicate content than missing content.

Author: I reworded it to put more focus on when the alt text was necessary.

Do you think we should make `alt` a required prop and remove the `""` default? In that case developers will need to explicitly pass `""` if their images are presentational.

Reviewer: I think that's a good idea. It forces developers to make a conscious decision.

> The Avatar can receive alt text like any other image element. However, in many cases, alt text in this context will not make sense.

Reviewer: ```suggestion
The Avatar can receive alt text like any other image element. However, in some cases, it might be better to omit the alt text.
```

For example, if the Avatar is used as an illustrative element in a products list next to each product's name, using the product name as alt text will be redundant: assistive technology will already read out the names once.

Therefore, the alt prop is not required and defaults to `""`, effectively making the Avatar invisible to assistive technology.

> Howver, bear in mind that the alt text is fundamental for accessibility if the Avatar is used without textual elements, for example as part of a products grid where only the images are shown.

Reviewer: ```suggestion
However, bear in mind that the alt text is fundamental for accessibility if the Avatar is used without textual elements, for example as part of a grid of products where only the images are shown.
```
